cluster/gce/gci/health-monitor.sh

```diff
@@ -24,11 +24,22 @@ set -o pipefail
 
 # We simply kill the process when there is a failure. Another systemd service will
 # automatically restart the process.
-function docker_monitoring {
+function container_runtime_monitoring {
+  # Container runtime startup takes time. Make initial attempts before starting
+  # killing the container runtime.
+  local -r initial_attempts=5
+  local attempt=1
+  local -r crictl="${KUBE_HOME}/bin/crictl"
+  local -r container_runtime="${CONTAINER_RUNTIME_NAME:-"docker"}"
+  until timeout timeout 60 "${crictl}" pods > /dev/null || (( attempt == initial_attempts ))
```

Comment:
Personally, I think it would be best to move the check for reaching max attempts inside the loop, in which case you could actually print a message saying you're giving up before `break`ing out of the loop...

Yes, I think in fact that might be critical, otherwise if you reach max attempts you will just proceed with the code. Is that really correct?

Well, maybe fallthrough is what you want, but I think at least logging a message about it would be good, then it's clear what happened here.

Reply:
> Yes, I think in fact that might be critical, otherwise if you reach max attempts you will just proceed with the code. Is that really correct?

Yeah, that is what we want. The logic here is just to make sure we give docker sometime to startup, before we start killing it (similar with the `sleep 120` below for kubelet).

I can move the attempt inside.

Reply:
Done.